### DISCUSSION TITLE
Your app can now provide Quick Look file previews on macOS!

### DISCUSSION THREAD

**Discussion by jtbandes** [atom/atom#22581] [Show and tell]
Hi everyone,

I recently created a package that you can use to add previews for custom file types on macOS 🎉

[quicklookjs](https://github.com/jtbandes/quicklookjs) is a macOS Quick Look Previewing extension that displays a web page. All you need to do is replace this web page with your own custom preview, and re-package the extension with your app. The preview page uses JavaScript to access the file data — allowing you to leveraging your existing web expertise and build tooling.

You'll find more instructions in [the README](https://github.com/jtbandes/quicklookjs). I hope you can make some cool apps with this – please reach out with any questions, or just to let me know what you've created! 😄

![](https://user-images.githubusercontent.com/14237/121449575-cb86f080-c935-11eb-9a45-b5837d517616.gif)

[Here's an example](https://github.com/foxglove/studio/pull/1187) of how I integrated it with our Electron app, Foxglove Studio. It looks like this:

https://user-images.githubusercontent.com/14237/121414564-7b426b00-c902-11eb-8088-81f22a8712d9.mov

**Comment by ThatXliner** [comment]
## ❗ I have an idea!

What if we added that into Atom? Atom *already* generates HTML with the syntax highlighting of your favorite theme and all. So why not just make a quick look with that?

**Reply by jtbandes** [reply]
That sounds cool! I know that some third-party plugins already exist to provide syntax highlighting on code, for instance: https://github.com/anthonygelibert/QLColorCode https://whomwah.github.io/qlstephen/ But having it built into an editor is nice because it doesn't require extra installation. It does mean the JS code to parse the language grammar and colorize the text would need to be bundled into the preview page.

I'm guessing the preview page would need to access the user's theme preferences, which are presumably stored in a file or NSUserDefaults somewhere. Since the preview extension is sandboxed, it can't necessarily access arbitrary files on disk. But I think it could be easily extended to support NSUserDefaults, and I'm sure we could find some kind of workaround for other sandboxing issues. I'd be happy to discuss these issues more if someone wants to try and make use of the project.
